Automattic/wp-calypso
# Themes: Add data-layer middleware for theme filters

No visual changes.

Adds [data-layer middleware](https://github.com/Automattic/wp-calypso/blob/master/client/state/data-layer/README.md) for new theme-filters endpoint (see D4901-code) that will eventually replace the [hardcoded theme taxonomies](https://github.com/Automattic/wp-calypso/blob/2646a3737315d23db72b5e7aa3135dd2bdc29972/client/my-sites/themes/theme-filters.js#L25) data.

This is one part of a set of changes that will allow theme filter terms to be updated only at the server with no Calypso changes required.
<img width="931" alt="screen shot 2017-04-03 at 15 45 25" src="https://cloud.githubusercontent.com/assets/7767559/24614905/a046873e-1884-11e7-8762-bbc4b096a70b.png">

**To Test**
* Check that `dispatch( { type:'THEME_FILTERS_REQUEST' } )` in the js console triggers network request
* Use redux devtools to check the payload of the `THEME_FILTERS_ADD` action type, which is dispatched on receipt of the API response:
<img width="477" alt="screen shot 2017-04-04 at 14 09 47" src="https://cloud.githubusercontent.com/assets/7767559/24658210/f3bf4fba-1940-11e7-8ed5-488d13ba6290.png">
* Generate an API response error by breaking the ssh tunnel to D4901-code and check that the error notice is displayed:
<img width="529" alt="screen shot 2017-04-04 at 14 07 53" src="https://cloud.githubusercontent.com/assets/7767559/24658254/1ccbd54a-1941-11e7-9393-61b724f42115.png">
